File: src/t92_ReverseLinkedListII.java
public class t92_ReverseLinkedListII {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        if (right == 1)
            return head;
        ListNode dummy = new ListNode();
        ListNode l = dummy;
        ListNode r;
        dummy.next = head;

        for (int i = 0; i < left - 1; i++)
            l = l.next;
        r = l.next;
        for (int i = left; i < right; i++) {
            ListNode rev = r.next;
            r.next = rev.next;
            rev.next = l.next;
            l.next = rev;
        }
        return dummy.next;
    }
}

File: src/python/t92_ReverseLinkedListII.py
def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
    if right == 1:
        return head
    dummy = ListNode()
    l: ListNode = dummy
    r: ListNode
    dummy.next = head

    for i in range(0, left - 1):
        l = l.next
    r = l.next
    for i in range(left, right):
        rev: ListNode = r.next
        r.next = rev.next
        rev.next = l.next
        l.next = rev
    return dummy.next
